src/app/BUILD.gn

```diff
@@ -150,6 +150,18 @@ source_set("test-event-trigger") {
   sources = [ "TestEventTriggerDelegate.h" ]
 }
 
+source_set("timer-delegates") {
+  sources = [
+    "TimerDelegates.cpp",
+    "TimerDelegates.h",
```

Comment:
What are these?

Looking at the content, it looks like they specifically only implement reporting::ReeportScheduler::TimerDelegate and specifically contain a `DefaultTimerDelegate`

Should they maybe reside in `app/reporting` and be named `DefaultTimerDelegate` instead? There is no plural delegates right now and the placement in app is unclear to me. Or alternatively, if this is specifically for icd, should these reside in ICD? Are there plans to add more to this which would make `Default` in the name inappropriate?

Reply:
Removed this change. I had doen this initially solve a dependency problem. In the end, i was able to remove the full app dependecy from the ICD tests.